### PR TITLE
Implement WordCounter and WordAggregator

### PR DESCRIPTION
**Pull request by joan-miralles**
Exercise: https://pro.codely.tv/library/introduccion-a-scala/63278/path/step/33533069/

### CODE REVIEW

File: src/test/tv/codely/scala_intro_examples/lesson_08_collections/WordCounterSpec.scala [hunk 1]
```diff
@@ -10,13 +10,18 @@ final class WordCounterSpec extends WordSpec with Matchers {
...
-    // Change this "ignore" keyword for the "in" one and make this test pass!
-    "return the number of words given a non empty text" ignore {
+    "return the number of words given a non empty text" in {
       val counter = new WordCounter
       val nonEmptyText = "Hi all"
       val wordsNumber = 2
 
+      counter.countWords(nonEmptyText) shouldBe wordsNumber
+    }
+    "return the number of words given a non empty text with spaces" in {
```
Comment (JavierCane) [inline src/test/tv/codely/scala_intro_examples/lesson_08_collections/WordCounterSpec.scala]: Good one!

Suggestion: Add other corner cases such as empty strings (0), punctuation, and so on 😬

File: src/main/tv/codely/scala_intro_examples/lesson_08_collections/WordCounter.scala [hunk 2]
```diff
@@ -1,9 +1,10 @@
 package tv.codely.scala_intro_examples.lesson_08_collections
 
 final class WordCounter {
-  // @ToDo: Here you have your exercise!
-  // @see tv.codely.scala_intro_examples.lesson_08_collections.WordCounterSpec
   def countWords(text: String): Int = {
-    0
+    if (text.isEmpty)
+      return 0
```
Comment (JavierCane) [inline src/main/tv/codely/scala_intro_examples/lesson_08_collections/WordCounter.scala]: `return` statement not needed. In fact, it's not Scala idiomatic to specify such clauses since we will always return the last statement in a code block. In order to remove it, we should add the `else` part of the `if` 🙂

File: src/main/tv/codely/scala_intro_examples/lesson_08_collections/WordAggregator.scala [hunk 3]
```diff
@@ -1,9 +1,15 @@
 package tv.codely.scala_intro_examples.lesson_08_collections
 
 final class WordAggregator {
-  // @ToDo: Here you have your exercise!
-  // @see tv.codely.scala_intro_examples.lesson_08_collections.WordAggregatorSpec
   def aggregateWords(text: String): Map[String, Int] = {
-    Map.empty
+    if (text.isEmpty)
+      return Map.empty
```
Comment (JavierCane) [inline src/main/tv/codely/scala_intro_examples/lesson_08_collections/WordAggregator.scala]: Same as with the `WordCounter` case. You don't need to explicitly declare the `return` statements 🙂

File: src/main/tv/codely/scala_intro_examples/lesson_08_collections/WordAggregator.scala [hunk 4]
```diff
@@ -1,9 +1,15 @@
...
+    var occurrences = Map[String, Int]()
```
Comment (JavierCane) [inline src/main/tv/codely/scala_intro_examples/lesson_08_collections/WordAggregator.scala]: Try to think about an alternative implementation without requiring mutability (`var`) 🙂

Hint: Instead of iterating over the `words: Array[String]` with a `foreach`, there're other methods which allow to accumulate a result between each array item iteration. Take a look at [the `scala.Array` documentation 😬](https://www.scala-lang.org/api/2.12.4/scala/Array.html)

**Review by JavierCane — APPROVED** [review]
Some suggestions added 🙂

@joan-miralles tell me if you want to apply these suggestions in this very same PR. If so, I would wait for them before merging it.

**Comment by joan-miralles** [comment]
Fixed! I've moved solution to exercise_solutions folder. (better if you want merge PR to public repository)

**Comment by JavierCane** [comment]
Thanks for moving your solution proposal to its own package. It definitely help us out!

Merging!
